File: visualiser/samples/javascript/calculator.js
import { Circle, Rectangle, Triangle } from './shapes.js';
import { roundResult } from './utils.js';
export class AreaCalculator {
  constructor() { this._shapes = []; }
  addCircle(radius)           { this._shapes.push(new Circle(radius)); }
  addRectangle(width, height) { this._shapes.push(new Rectangle(width, height)); }
  addTriangle(base, height)   { this._shapes.push(new Triangle(base, height)); }
  totalArea() { return roundResult(this._shapes.reduce((s, sh) => s + sh.area(), 0)); }
  largest()   { return this._shapes.reduce((a, b) => a.area() > b.area() ? a : b, null); }
  report()    { return [...this._shapes.map(s => s.describe()), `Total: ${this.totalArea()}`].join('\n'); }
}

File: visualiser/samples/python/src/shapes/calculator.py
"""Area calculator."""
from src.shapes.shapes import Circle, Rectangle, Triangle
from src.shapes.utils import round_result

class AreaCalculator:
    def __init__(self) -> None:
        self._shapes: list = []
    def add_circle(self, radius: float) -> None:
        self._shapes.append(Circle(radius))
    def add_rectangle(self, width: float, height: float) -> None:
        self._shapes.append(Rectangle(width, height))
    def add_triangle(self, base: float, height: float) -> None:
        self._shapes.append(Triangle(base, height))
    def total_area(self) -> float:
        return round_result(sum(s.area() for s in self._shapes))
    def largest(self):
        return max(self._shapes, key=lambda s: s.area(), default=None)
    def report(self) -> str:
        lines = [s.describe() for s in self._shapes]
        lines.append(f"Total area: {self.total_area()}")
        return "\n".join(lines)
